pkg/vpa/service_test.go

```diff
@@ -135,3 +140,303 @@ func TestMakeAllVerticalContainerResourcePhaseRunning(t *testing.T) {
+						ContainerPolicies: []vpav1.ContainerResourcePolicy{
+							{
+								ContainerName: "app",
+								MinAllowed: v1.ResourceList{
+									v1.ResourceMemory: resource.MustParse("1Gi"),
+									v1.ResourceCPU:    resource.MustParse("1"),
+								},
+							},
+							{
+								ContainerName: "istio-proxy",
+								MinAllowed: v1.ResourceList{
+									v1.ResourceMemory: resource.MustParse("1Gi"),
+									v1.ResourceCPU:    resource.MustParse("1"),
+								},
+							},
+						},
+					},
+				},
+			},
+			wantTortoise: &autoscalingv1beta3.Tortoise{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "tortoise",
+					Namespace: "default",
+				},
+				Status: autoscalingv1beta3.TortoiseStatus{
+					AutoscalingPolicy: []autoscalingv1beta3.ContainerAutoscalingPolicy{
+						{
+							ContainerName: "app",
+							Policy: map[v1.ResourceName]v1beta3.AutoscalingType{
+								v1.ResourceMemory: v1beta3.AutoscalingTypeVertical,
+								v1.ResourceCPU:    v1beta3.AutoscalingTypeHorizontal,
+							},
+						},
+						{
+							ContainerName: "istio-proxy",
+							Policy: map[v1.ResourceName]v1beta3.AutoscalingType{
+								v1.ResourceMemory: v1beta3.AutoscalingTypeVertical,
+								v1.ResourceCPU:    v1beta3.AutoscalingTypeHorizontal,
+							},
+						},
+					},
+					ContainerResourcePhases: []autoscalingv1beta3.ContainerResourcePhases{
+						{
+							ContainerName: "app",
+							ResourcePhases: map[v1.ResourceName]autoscalingv1beta3.ResourcePhase{
+								v1.ResourceCPU: {
+									Phase: autoscalingv1beta3.ContainerResourcePhaseGatheringData,
+								},
+								v1.ResourceMemory: {
+									Phase: autoscalingv1beta3.ContainerResourcePhaseGatheringData,
+								},
+							},
+						},
+						{
+							ContainerName: "istio-proxy",
+							ResourcePhases: map[v1.ResourceName]autoscalingv1beta3.ResourcePhase{
+								v1.ResourceCPU: {
+									Phase: autoscalingv1beta3.ContainerResourcePhaseGatheringData,
+								},
+								v1.ResourceMemory: {
+									Phase: autoscalingv1beta3.ContainerResourcePhaseGatheringData,
+								},
+							},
+						},
+					},
+				},
+				Spec: autoscalingv1beta3.TortoiseSpec{
+					ResourcePolicy: []autoscalingv1beta3.ContainerResourcePolicy{
+						{
+							ContainerName: "app",
+							MinAllocatedResources: v1.ResourceList{
+								v1.ResourceMemory: resource.MustParse("1Gi"),
+								v1.ResourceCPU:    resource.MustParse("1"),
+							},
+						},
+						{
+							ContainerName: "istio-proxy",
+							MinAllocatedResources: v1.ResourceList{
+								v1.ResourceMemory: resource.MustParse("1Gi"),
+								v1.ResourceCPU:    resource.MustParse("1"),
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c := &Service{
+				c:        fake.NewSimpleClientset(tt.initialVPA), // import "k8s.io/autoscaler/vertical-pod-autoscaler/pkg/client/clientset/versioned/fake".
```

Comment:
no need to keep this comment.

```suggestion
				c:        fake.NewSimpleClientset(tt.initialVPA),
```